File: ejercicio02.js
// Archivo: ejercicio02.js

console.log("--- 1. OPERADORES ---");

// Aritméticos
let a = 10, b = 3;
console.log("Aritméticos (10 % 3): " + (a % b));

// Asignación
a += 5; 
console.log("Asignación (a += 5): " + a);

// Comparación (igualdad débil vs estricta)
console.log("Comparación (15 == '15'): " + (15 == '15')); // true
console.log("Identidad estricta (15 === '15'): " + (15 === '15')); // false

// Lógicos
console.log("Lógicos (true && false): " + (true && false));

// Bits
console.log("Bits (10 & 3): " + (10 & 3));

// Pertenencia (in)
const usuario = { nombre: "Gemini", rol: "IA" };
console.log("Pertenencia ('nombre' in usuario): " + ('nombre' in usuario));

// --- 2. ESTRUCTURAS DE CONTROL ---
console.log("\n--- ESTRUCTURAS DE CONTROL ---");

// Condicionales
let edad = 18;
if (edad >= 18) {
    console.log("Condicional: Eres mayor de edad.");
}

switch (a) {
    case 15: console.log("Switch: El valor es 15."); break;
    default: console.log("Switch: Otro valor.");
}

// Iterativas
console.log("For loop:");
for (let i = 0; i < 3; i++) console.log("  i = " + i);

console.log("While loop:");
let contador = 0;
while (contador < 2) {
    console.log("  Contando: " + contador);
    contador++;
}

// Excepciones (try-catch-finally)
try {
    throw new Error("Algo salió mal");
} catch (error) {
    console.log("Excepción capturada: " + error.message);
} finally {
    console.log("Bloque finally: Siempre se ejecuta.");
}

File: ejercicio02.java
public class ejercicio02 {
    public static void main(String[] args) {
        
        // --- 1. OPERADORES ---
        System.out.println("--- EJEMPLOS DE OPERADORES ---");
        
        int a = 10, b = 3;
        
        // Aritméticos
        System.out.println("Aritméticos (10 + 3): " + (a + b));
        
        // Asignación
        a += 5; // a ahora es 15
        System.out.println("Asignación (a += 5): " + a);
        
        // Comparación
        System.out.println("Comparación (15 > 3): " + (a > b));
        
        // Lógicos
        System.out.println("Lógicos (true && false): " + (true && false));
        
        // Bits (AND a nivel de bit)
        System.out.println("Bits (10 & 3): " + (10 & 3));
        
        // Identidad (referencia de objetos)
        String s1 = "Hola";
        String s2 = "Hola";
        System.out.println("Identidad (s1 == s2): " + s1.equals(s2)); // true (mismo literal)
        System.out.println("Igualdad de contenido (s1.equals(s2)): " + s1.equals(s2)); // true

        // --- 2. ESTRUCTURAS DE CONTROL ---
        System.out.println("\n--- ESTRUCTURAS DE CONTROL ---");

        // Condicionales
        int edad = 18;
        if (edad >= 18) {
            System.out.println("Condicional: Eres mayor de edad.");
        }
        
        switch (a) {
            case 15 -> System.out.println("Switch: El valor es 15.");
            default -> System.out.println("Switch: Otro valor.");
        }

        // Iterativas
        System.out.print("For loop: ");
        for (int i = 0; i < 3; i++) System.out.print(i + " ");
        
        System.out.println("\nWhile loop: ");
        int contador = 0;
        while (contador < 2) {
            System.out.println("  Contando: " + contador);
            contador++;
        }

        // Excepciones
        try {
        } catch (ArithmeticException e) {
            System.out.println("Excepción capturada: División por cero.");
        } finally {
            System.out.println("Bloque finally: Esto siempre se ejecuta.");
        }
    }
}